packages/shorebird_cli/lib/src/commands/patch_new/android_patcher.dart

```diff
@@ -0,0 +1,177 @@
+import 'package:crypto/crypto.dart';
+import 'package:io/io.dart';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/archive_analysis/android_archive_differ.dart';
+import 'package:shorebird_cli/src/archive_analysis/archive_differ.dart';
+import 'package:shorebird_cli/src/artifact_builder.dart';
+import 'package:shorebird_cli/src/artifact_manager.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/commands/patch_new/patcher.dart';
+import 'package:shorebird_cli/src/commands/release_new/release_type.dart';
+import 'package:shorebird_cli/src/doctor.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/platform/platform.dart';
+import 'package:shorebird_cli/src/shorebird_android_artifacts.dart';
+import 'package:shorebird_cli/src/shorebird_flutter.dart';
+import 'package:shorebird_cli/src/shorebird_validator.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+
+/// {@template android_patcher}
+/// Functions to create an Android patch.
+/// {@endtemplate}
+class AndroidPatcher extends Patcher {
+  /// {@macro android_patcher}
+  AndroidPatcher({required super.flavor, required super.target});
+
+  @override
+  ReleaseType get releaseType => ReleaseType.android;
+
+  @override
+  String get primaryReleaseArtifactArch => 'aab';
+
+  @override
+  ArchiveDiffer get archiveDiffer => AndroidArchiveDiffer();
+
+  @override
+  Future<void> assertArgsAreValid() async {
+    // Nothing to do here
+  }
+
+  @override
+  Future<void> assertPreconditions() async {
+    try {
+      await shorebirdValidator.validatePreconditions(
+        checkUserIsAuthenticated: true,
+        checkShorebirdInitialized: true,
+        validators: doctor.androidCommandValidators,
+      );
+    } on PreconditionFailedException catch (e) {
+      exit(e.exitCode.code);
+    }
```

Comment:
Should the caller do this?

Reply:
The general contract in the corresponding release commands is that the functions themselves handle logging and exiting (if necessary) and should not throw.